runtimes/dart-2.18/src/function_types.dart

```diff
@@ -14,6 +13,39 @@ class RuntimeRequest {
   String queryString;
   Map<String, String> query;
 
+  dynamic get body {
+    String contentType = (this.headers['content-type'] ?? 'text/plain').toLowerCase();
+
+    if(contentType.startsWith("application/json")) {
+      if(this.bodyBinary.isEmpty) {
+        return new Map<String, dynamic>();
+      } else {
+        return this.bodyJson;
+      }
+    }
+
+    List<String> binaryTypes = ["application/", "audio/", "font/", "image/", "video/"];
+    for(final binaryType in binaryTypes) {
+      if(contentType.startsWith(binaryType)) {
+        return this.bodyBinary;
+      }
+    }
+
+    return this.bodyText;
+  }
+
+  String get bodyRaw {
+    return this.bodyText;
```

Comment:
why does bodyRaw return the parsed text? bodyRaw is supposed to be the unprocessed binary data right ?

Reply:
`bodyRaw` is not supposed to exist, we only keep it for backwards compatibility. and in V3, `bodyRaw` was string, because we had wrong assumption that bodies are always string.